src/data/datainterfaces.ts

```diff
@@ -15,152 +82,388 @@ export interface IFieldDefinition {
+  // Extends IFieldConfigDefault
+  ...IFieldConfigDefaultObject,
+};
+
+export const IInstanceTitleFieldConfigSchema = Joi.object(IInstanceTitleFieldConfigObject);
+
 export interface ILabelConfig extends IFieldConfig {
   labelHtml?: string;
 }
 
+const ILabelConfigObject: ILabelConfig = {
+  labelHtml: (Joi.string().allow("") as unknown) as string,
+  // Extends IFieldConfig
+  ...IFieldConfigObject,
+};
+
+export const ILabelConfigSchema = Joi.object(ILabelConfigObject);
+
 export type IMailFieldConfig = IFieldConfigDefault;
 
+export const IMailFieldConfigSchema = Joi.object(IFieldConfigDefaultObject);
+
 export interface INumberFieldConfig extends IFieldConfigDefault {
   evalDefaultValue: boolean;
   onlyIntegers: boolean;
 }
 
+const INumberFieldConfigObject: INumberFieldConfig = {
+  evalDefaultValue: (Joi.boolean().required() as unknown) as boolean,
+  onlyIntegers: (Joi.boolean().required() as unknown) as boolean,
+  // Extends IFieldConfigDefault
+  ...IFieldConfigDefaultObject,
+};
+
+export const INumberFieldConfigSchema = Joi.object(INumberFieldConfigObject);
+
 export interface IRadioButtonGroupEntry {
   name: string;
   value: number;
   selected: boolean;
 }
 
+const IRadioButtonGroupEntryObject: IRadioButtonGroupEntry = {
+  name: (Joi.string().allow("").required() as unknown) as string,
+  value: (Joi.number().required() as unknown) as number,
+  selected: (Joi.boolean().required() as unknown) as boolean,
+};
+
+export const IRadioButtonGroupEntrySchema = Joi.object(IRadioButtonGroupEntryObject);
+
 export interface IRadioButtonFieldConfig extends IFieldConfigDefault {
   entries: IRadioButtonGroupEntry[];
   oneEntryMustBeChecked: boolean;
 }
 
+const IRadioButtonFieldConfigObject: IRadioButtonFieldConfig = {
+  entries: (Joi.array().items(Joi.object(IRadioButtonGroupEntryObject)).required() as unknown) as IRadioButtonGroupEntry[],
+  oneEntryMustBeChecked: (Joi.boolean().required() as unknown) as boolean,
+  // Extends IFieldConfigDefault
+  ...IFieldConfigDefaultObject,
+};
+
+export const IRadioButtonFieldConfigSchema = Joi.object(IRadioButtonFieldConfigObject);
+
 export interface IRiskAssessmentFieldConfig extends IFieldConfig {
   validationExpression: string | undefined;
 }
 
+const IRiskAssessmentFieldConfigObject: IRiskAssessmentFieldConfig = {
+  validationExpression: (Joi.string().allow("") as unknown) as string,
+  // Extends IFieldConfig
+  ...IFieldConfigObject,
+};
+
+export const IRiskAssessmentFieldConfigSchema = Joi.object(IRiskAssessmentFieldConfigObject);
+
 export interface IRoleOwnerFieldConfig extends IFieldConfig {
   validationExpression?: string;
   defaultValue: "NoValue" | "CurrentUser" | undefined;
 }
 
+const IRoleOwnerFieldConfigObject: IRoleOwnerFieldConfig = {
+  validationExpression: (Joi.string().allow("") as unknown) as string,
+  defaultValue: (Joi.string().pattern(new RegExp("^NoValue$|^CurrentUser$")) as unknown) as "NoValue" | "CurrentUser",
```

Comment:
I would prefer if we could find a test for this as I think it is basically the same case as with the enums.